### PR TITLE
Split out arrow-csv (#2594)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #2594

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-csv/src/reader.rs [hunk 1]
```diff
@@ -1131,436 +1131,11 @@ impl ReaderBuilder {
...
-    use std::io::{Cursor, Write};
+    use std::io::Write;
     use tempfile::NamedTempFile;
 
-    use crate::array::*;
-    use crate::compute::cast;
-    use crate::datatypes::Field;
     use chrono::prelude::*;
 
-    #[test]
```
Comment (tustvold) [inline arrow-csv/src/reader.rs:1143]: These are moved to an integration test in the arrow crate

File: arrow-csv/src/writer.rs [hunk 2]
```diff
@@ -595,105 +590,6 @@ sed do eiusmod tempor,-556132.25,1,,2019-04-18T02:45:55.555000000,23:46:03,foo
         );
     }
 
-    #[cfg(feature = "chrono-tz")]
```
Comment (tustvold) [inline arrow-csv/src/writer.rs:598]: Moved to integration test in arrow crate

File: arrow/Cargo.toml [hunk 3]
```diff
@@ -265,5 +264,9 @@ harness = false
 required-features = ["test_utils"]
 
 [[test]]
-name = "ipc_integration"
+name = "ipc"
```
Comment (tustvold) [inline arrow/Cargo.toml:267]: Opted to remove the redundant integration suffix

File: arrow-csv/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "arrow-csv"
+version = "26.0.0"
+description = "Support for the Arrow CSV format"
```
Comment (alamb) [inline arrow-csv/Cargo.toml:21]:
```suggestion
description = "Support for parsing CSV format into the Arrow format"
```

File: arrow-csv/src/reader.rs [hunk 5]
```diff
@@ -22,11 +22,11 @@
 //!
 //! Example:
 //!
-//! ```
-//! use arrow::csv;
-//! use arrow::datatypes::{DataType, Field, Schema};
-//! use std::fs::File;
-//! use std::sync::Arc;
+//! ```no_run
```
Comment (alamb) [inline arrow-csv/src/reader.rs:25]: Why is this changed to "no_run"? It seems like it is a good thing to keep testing, no?

**Review by alamb — APPROVED** [review]
Looks good to me

File: arrow-csv/src/reader.rs (hunk 5 above)
Comment (tustvold) [reply arrow-csv/src/reader.rs:25]: Because the file it reads is no longer part of this crate
